### PR TITLE
Ability to set FeeGrantAddress when signing transactions

### PR DESCRIPTION
**Pull request by felipead**
## Purpose of Changes and their Description

Set FeeGrantAddress when signing reputer and worker/forecaster transactions

## Link(s) to Ticket(s) or Issue(s) resolved by this PR

ENGN-3546

https://linear.app/alloralabs/issue/ENGN-3546/offchain-node-utilize-xfeegrant-module-in-workersreputersforecasters

## Are these changes tested and documented?

- [x] If tested, please describe how. If not, why tests are not needed.
- [x] If documented, please describe where. If not, describe why docs are not needed.
- [x] Added to `Unreleased` section of `CHANGELOG.md`?

--

Documented the go code

### CODE REVIEW

File: lib/transaction/builder_test.go [hunk 1]
```diff
@@ -74,6 +74,7 @@ func TestBuildAndSignTransactionWithDifferentParams(t *testing.T) {
 					OverrideFees:  0,
 					SimulateTx:    false,
 				},
+				FeeGranterAddress: "",
```
Comment (spooktheducks) [inline lib/transaction/builder_test.go:77]: Is there a reasonably efficient way to write a test for a non-empty fee granter?

**Review by spooktheducks — CHANGES_REQUESTED** [review]

File: lib/transaction/builder_test.go (hunk 1 above)
Comment (felipead) [reply lib/transaction/builder_test.go:77]: Added two test cases:

- Using a fee granter address
- Edge case - invalid fee granter address

**Review by xmariachi — APPROVED** [review]
